### PR TITLE
add a factor to scale fatigue up and downhill

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- pls no bully, baer
- renamed `_velocity` to `_speed` so it's not confused with the `velocity` command I use for this and one is not inclined to replace it.
- loosely based on this graph:

### CODE REVIEW

**Comment by commy2** [comment]
`surfaceNormal` doesn't change whether you go up or down though.
good point about `getPosATL` and pathway lods

Currently this goes to zero if you walk along the slope, which we don't want.
I'll be waiting for baers feedback before doing anything though.

**Comment by esteldunedain** [comment]
> doesn't change whether you go up or down though.

```sqf
_slope = (surfaceNormal _pos) vectorProduct (vectorNormalized velocity ACE_player);
```
Vector algebra 😄

**Comment by esteldunedain** [comment]
> Currently this goes to zero if you walk along the slope, which we don't want.

I think that's not bad. But if you really want to take into account that walking along a slope is a bit more tiresome than walking on a flat surface you could add a coefficient based on `1 - (surfaceNormal _pos select 2)` that's independent of walking direction.

**Comment by BaerMitUmlaut** [comment]
Don't quite see the need for renaming `_velocity`, there's also a `speed` command after all. You've also kept in the old slope penalty as it is right now, that should be changed.

**Comment by commy2** [comment]
>You've also kept in the old slope penalty as it is right now, that should be changed.

What part of this is the "old slope penalty"? `_terrainGradient` ? Of course that should stay, because the surface normal isn't part of my factor.

**Comment by commy2** [comment]
>Don't quite see the need for renaming _velocity, there's also a speed command after all.

I just think it's weird to use `velocity` when you already have a variable named `velocity`. Usually it's like this: Absolute value -> speed, Vector -> velocity (that's what the commands return, except that speed is using the 3.6 factor for km/h)
This is totally unofficial commy SQF variable naming though.

**Comment by BaerMitUmlaut** [comment]
`_terrainGradient` was the main factor that penalized slopes which provided decent results. Your factor adds another penalty, but without changing the `_terrainGradient` factor - that punishes the player too much, because you're essentially punishing him twice.

**Comment by commy2** [comment]
`_terrainAltitudeFactor` is always smaller or equal than 1. It's supposed to make walking downwards easier.   You said walking uphills should stay the same.

**Comment by BaerMitUmlaut** [comment]
That doesn't work though. For example, let's say I walk parallel to a slope.
```
private _terrainAltitudeFactor = 2 * abs _terrainAltitudeDelta min 1;
```
In that case `_terrainAltitudeDelta` would be 0 because I don't actually climb, so `_terrainAltitudeFactor` would then also be 0. Later you do
```
_terrainGradient * _terrainAltitudeFactor
```
which would then also become 0, so no penalty for walking on slopes, which is the opposite of what we want.

**Comment by commy2** [comment]
Wanna trade with the weather issue report?

**Comment by BaerMitUmlaut** [comment]
Sure :smile:

File: addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf [hunk 1]
```diff
@@ -32,6 +32,14 @@ private _gearMass = ((loadAbs _unit + _virtualLoad) * 0.1 / 2.2046) * GVAR(loadF
 private _terrainFactor = 1;
```
Comment (ulteq) [inline addons/advanced_fatigue/functions/fnc_getMetabolicCosts.sqf]: `_terrainFactor` -> dead code?

**Pull request closed without merging**